bit9PlatformAPI/examples/connectors/Metascan.py

```diff
@@ -0,0 +1,247 @@
+        if fileHash in self.scheduledScans.keys():
+            lastAttempt = self.scheduledScans[fileHash]
+            # Be polite and don't keep asking MS for status too often. If we already tried recently, bail
+            if lastAttempt['nextCheck'] > datetime.datetime.now():
+                return
+            # Get our dataId we got from MS last time around
+            print("Now attempting to send the data ID")
+            dataId = lastAttempt['data_id']
+            r = requests.get(self.ms_url + "/file/" + dataId, params={'':''})
+            print("This is the URL being sent %s" % (self.ms_url + "/file/" + dataId))
+            print(r)
+        else:
+            # we have not asked MS yet. Try with file hash
+            r = requests.get(self.ms_url + "/hash/" + fileHash, params={'':''})
+
+        r.raise_for_status()
+        if r.status_code == 204:
+            # no results from MS because rate limit was reached. We will try again later.
+            return
+
+        scanResults = r.json()
+        x=scanResults.get("scan_results",{})
+        # Check if we got results...
+        self.msresults=x.get('scan_all_result_i')
+        self.msavtotal=x.get('total_avs')
+        if self.msresults is not None:
+            # Yes, MS has them. Report results and we are done with the file
+            self.reportResultToBit9(pa['id'], scanResults)
+            dataId =  None
+        elif scanResults.get('data_id') is not None:
+            # MS already knows about the file, but scan is not complete. We got dataId for future reference
+            dataId = scanResults['data_id'];
+        elif pa['uploaded'] == 1:
+            # We have file and now we will upload it to MS
+            dataId = self.uploadFileToMS(pa)
+        else:
+            # if we end here, it means that MS doesn't have file, and Bit9 hasn't uploaded it yet from the agent
+            # we will come back again in 30 seconds
+            dataId = None
+
+        if dataId is not None:
+            # Remember dataId since MS wants use to use it for future references to the file
+            # We will try again in 1 hour
+            self.scheduledScans[fileHash] = {'fileID': dataId, 'nextCheck': datetime.datetime.now()
+                                                                      + datetime.timedelta(0, 3600)}
+
+# -------------------------------------------------------------------------------------------------
+# Main body of the script
+
+bit9 = bit9api.bit9Api(
+    "https://yourbit9serverurl.domain.com",  # Replace with actual Bit9 server URL
+    token="",  # Replace with actual Bit9 user token for integration
+    ssl_verify=False  # Don't validate server's SSL certificate. Set to True unless using self-signed cert on IIS
+)
+
+msConnector = metascanConnector(
+    bit9,
+    ms_token='123456',  # Replace with your VT key
+    download_location= "C:\YourDownloadLocation"  # Replace with actual local file location. If not set,
```

Comment:
For Windows paths you will want to use a Python raw string literal `r"c:\YourDownloadLocation"` so it won't interpret `"\X"` as an escape character. Add `r` before the first quote character.